test/lib/deployment/test-deployment.js

```diff
@@ -283,7 +283,9 @@ async function runProbe(probe, deploymentId, deploymentUrl, ctx) {
       }
     });
     hadTest = true;
-  } else if (probe.notResponseHeaders) {
+  }
+
+  if (probe.notResponseHeaders) {
```

Comment:
Unrelated but useful, this just ensures that if the probe specified both `responseHeaders` AND `notResponseHeaders` it checks both of them.